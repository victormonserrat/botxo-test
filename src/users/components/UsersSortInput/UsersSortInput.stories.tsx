import { action } from "@storybook/addon-actions";
import React from "react";
import UsersSortInput from "./UsersSortInput";

export default {
  title: "Users Sort input",
  component: UsersSortInput,
};

export const Default = () => <UsersSortInput onChange={action("clicked")} />;

export const Selected = () => (
  <UsersSortInput criteria={"firstName"} onChange={action("clicked")} />
);
