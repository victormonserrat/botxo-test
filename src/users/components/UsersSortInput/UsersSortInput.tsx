import React from "react";
import { Dropdown } from "semantic-ui-react";
import { UsersSortCriteria } from "../../models";

export interface Props {
  criteria?: UsersSortCriteria;
  onChange: (criteria: UsersSortCriteria) => void;
}

export const UsersSortInput: React.FunctionComponent<Props> = ({
  criteria,
  onChange,
}: Props) => {
  const options = [
    { key: 0, text: "First name", value: "firstName" },
    { key: 1, text: "Last name", value: "lastName" },
    { key: 2, text: "Age", value: "age" },
    { key: 3, text: "Connections", value: "connections" },
  ];

  return (
    <Dropdown
      item
      options={options}
      text={
        (criteria === "firstName" && "First Name") ||
        (criteria === "lastName" && "Last Name") ||
        (criteria === "age" && "Age") ||
        (criteria === "connections" && "Connections") ||
        "Order by"
      }
      onChange={(event, { value }) => {
        onChange(value as UsersSortCriteria);
      }}
    />
  );
};

export default UsersSortInput;
