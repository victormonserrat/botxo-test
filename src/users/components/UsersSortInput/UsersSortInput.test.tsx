import { mount, shallow } from "enzyme";
import React from "react";
import { Dropdown, DropdownItem } from "semantic-ui-react";
import UsersSortInput from "./UsersSortInput";

describe("UserCard", () => {
  const onChangeMock = jest.fn();

  it("should be rendered correctly", () => {
    const component = shallow(<UsersSortInput onChange={onChangeMock} />);

    expect(component).toMatchSnapshot();
  });

  it("should be unselected by default", () => {
    const wrapper = mount(<UsersSortInput onChange={onChangeMock} />);

    expect(wrapper.find(Dropdown).prop("text")).toBe("Order by");
  });

  it("can be changed", () => {
    const wrapper = mount(<UsersSortInput onChange={onChangeMock} />);

    wrapper.find(DropdownItem).first().simulate("click");

    expect(onChangeMock).toHaveBeenCalledWith("firstName");
  });
});
