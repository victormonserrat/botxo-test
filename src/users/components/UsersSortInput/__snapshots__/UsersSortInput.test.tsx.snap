// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard should be rendered correctly 1`] = `ShallowWrapper {}`;
